aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -67,6 +68,41 @@ void convert_indices_from_coo_to_csr_cuda(const Tensor& result, const Tensor& in
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
+template <typename input_t, typename output_t>
+__global__ void convert_indices_from_csr_to_coo_cuda_kernel(output_t* data_out, const input_t* data_in, const int64_t nrows) {
+  int64_t tid = blockDim.x * blockIdx.x + threadIdx.x;
+
+  if (tid < nrows) {
+    for (int64_t i = data_in[tid]; i < data_in[tid + 1]; i++)
+      data_out[i] = static_cast<output_t>(tid);
+  }
+}
+
+template <typename input_t, typename output_t>
+void convert_indices_from_csr_to_coo_cuda(const Tensor& result, const Tensor& crow_indices, const Tensor& col_indices) {
+  int64_t nrows = crow_indices.numel() - 1;
+  if (nrows == 0) {
+    result.zero_();
+    return;
+  }
+
+  auto crow_indices_ = crow_indices.expect_contiguous();
+  auto col_indices_ = col_indices.expect_contiguous();
+  const input_t* crow_indices_data_in = crow_indices_->data_ptr<input_t>();
+  const input_t* col_indices_data_in = col_indices_->data_ptr<input_t>();
+  TORCH_INTERNAL_ASSERT(result.is_contiguous());
+  output_t* data_out = result.data_ptr<output_t>();
+
+  // Run numel threads...
+  int64_t THREADS = at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;
+  int64_t BLOCKS = (nrows + THREADS) / THREADS;
+  at::cuda::CUDAStream stream = at::cuda::getCurrentCUDAStream();
+  at::cuda::ThrustAllocator allocator;
+  std::copy(col_indices_data_in, col_indices_data_in + col_indices.numel(), &data_out[col_indices.numel()]);
```

Comment:
Repeating the comment just for visibility, can we do ```result.copy_(col_indices)``` here?

Reply:
[Done.](https://github.com/pytorch/pytorch/pull/66774/commits/86488f45633c406d3220b86edb8febe2bbcc7e05#diff-38121953c8435aa8bafb16fb1a470fee958a1ff03a01f811e8d94371ae34b55d)